packages/mco/components/data-policies/drpolicy-actions/apply-policy/application-selector.tsx

```diff
@@ -0,0 +1,197 @@
+
+const areSomeDescendantsChecked = (dataItem, selectedNames) =>
+  dataItem.children ? dataItem.children.some(child => areSomeDescendantsChecked(child, selectedNames)) : isChecked(dataItem, selectedNames);
+
+const mapTree = (item: TreeViewDataItem, selectedNames: TreeViewDataItemMapping) => {
+  // Reset checked properties to be updated
+  item.checkProps.checked = false;
+  const hasCheck = areAllDescendantsChecked(item, selectedNames);
+  const hasPartialCheck = areSomeDescendantsChecked(item, selectedNames);
+  if (hasCheck) {
+    item.checkProps.checked = true;
+  } else if (hasPartialCheck) {
+    item.checkProps.checked = null;
+  }
+  if (item.children) {
+    return {
+      ...item,
+      children: item.children.map(child => mapTree(child, selectedNames))
+    };
+  }
+  return item;
+};
+
+const badge = (subNames: string[], plsRule: string) => {
+  return(
+    <>
+      {subNames.slice(0,subNames.length).map((subName) =>
+        <p key={subName}> {subName} </p>
+      )}
+      <TextContent>
+        <Text component={TextVariants.small}>{plsRule}</Text>
+      </TextContent>
+    </>
+  )
+};
+
+const filterApplications = (application: ApplicationKind, searchValue: string) => 
+      application?.metadata?.name.toLowerCase().includes(searchValue.toLowerCase());
+
+export const ApplicationSelector: React.FC<ApplicationSelectorProps> = (props) => {
+    const {applicationToPlacementRuleMapping, selectedNames, setSelectedNames} = props;
+    
+    const { t } = useTranslation('plugin__odf-console');
+    const [filteredApplicationToPlacementRuleMapping, setFilteredApplicationToPlacementRuleMapping] = React.useState<AppToPlacementRule>({});
+    const [options, setOptions] = React.useState<TreeViewDataItem[]>([]);
+    const [searchAppName, setSearchAppName] = React.useState("");
+
+    React.useEffect(() => {
+      setFilteredApplicationToPlacementRuleMapping(applicationToPlacementRuleMapping);
+    },[applicationToPlacementRuleMapping]);
+
+    React.useEffect(() => {
+      const tempItem: TreeViewDataItem[] = [];
+      Object.keys(filteredApplicationToPlacementRuleMapping).forEach((appUniqueName) => {
+        const childerns: TreeViewDataItem[] = [];
+        Object.keys(filteredApplicationToPlacementRuleMapping[appUniqueName]?.placements).forEach((placementUniqueName) => {
+          const placementName = filteredApplicationToPlacementRuleMapping[appUniqueName].placements[placementUniqueName];
+          const { subscriptions, placementRules } = placementName || {};
+          const subNames = subscriptions?.map((sub) => sub?.metadata?.name);
+          childerns.push({
+            name: badge(subNames, placementRules?.metadata?.name),
+            id: placementUniqueName,
+            checkProps: { checked: false },
+          });
+        });
+        tempItem.push({
+          name: filteredApplicationToPlacementRuleMapping?.[appUniqueName]?.application?.metadata?.name,
```

Comment:
lets use shorter -"map" at all places
```suggestion
          name: filteredApplicationToPlacementRuleMap?.[appUniqueName]?.application?.metadata?.name,
```